contributed_definitions/nyaml/NXmanipulator.yaml

```diff
@@ -13,32 +13,124 @@ NXmanipulator(NXobject):
   type(NX_CHAR):
     doc: |
       Type of manipulator, Hexapod, Rod, etc.
-  cryocoolant(NX_BOOLEAN):
+  cryostat(NXactuator):
     doc: |
-      Is cryocoolant flowing through the manipulator?
-  cryostat_temperature(NX_FLOAT):
-    unit: NX_TEMPERATURE
+      Cryostat for cooling the sample.
+    physical_quantity:
+      enumeration: [temperature]
+    (NXpid):
+      setpoint(NX_FLOAT):
+        unit: NX_TEMPERATURE
+        doc: |
+          In case of a fixed or averaged cooling temperature, this is the scalar temperature setpoint.
+          It can also be a 1D array of temperature setpoints (without time stamps).
+      setpoint_log(NXlog):
+        value(NX_FLOAT):
+          unit: NX_TEMPERATURE
+          doc: |
+            In the case of an experiment in which the temperature is changed and the setpoints are
+            recorded with time stamps, this is an array of length m of temperature setpoints.
+  temperature_sensor(NXsensor):
     doc: |
-      Temperature of the cryostat (coldest point)
-  heater_power(NX_FLOAT):
-    unit: NX_POWER
+      Temperature sensor measuring the sample temperature.
```

Comment:
seeing the need to state "sample" would it help to have a field in NXsensor and NXactuator whihc details which thing it is measuring or effecting respectively?
Semantically using the symbol "temperature_sensor" has no effect other to please human intuition about what to possibly find below here. That's why you have to tell in the docstring you measure the SAMPLE temperature and not just some temperature an attribute, thats exactly one such example where NeXus is not an as robust semantic way of doing things which in e.g. OWL you would have an extra relation to an object which this sensor is measuring in this case maybe a SENSOR -> takes temperature -> SAMPLE s.th.

Reply:
It is not a bad idea to link to the instance that is measured/effected in NXsensor and NXactuator. However, in this case, I have opted for using different instances of `NXenvrinment` on `NXmpes/NXsample` and then, inside these instances, there are actuators/sensors which are links to the devices defined in NXmanipulator. So, that link already exists (in the other direction than you are proposing here).

Maybe this is something for the future when these base classes are actually used,

Reply:
I see value of having examples for both so yes, let this one be such example for the other inference direction.